src/xenia/kernel/xam/xam_ui.cc

```diff
@@ -1176,16 +1178,60 @@ dword_result_t XamShowMarketplaceUI_entry(dword_t user_index, dword_t ui_type,
   cxxopts::OptionNames buttons;
 
   switch (ui_type) {
-    case 0:
+    case X_MARKETPLACE_ENTRYPOINT::ContentList:
       desc =
           "Game requested to open marketplace page with all content for the "
           "current title ID.";
       break;
-    case 1:
+    case X_MARKETPLACE_ENTRYPOINT::ContentItem:
       desc = fmt::format(
           "Game requested to open marketplace page for offer ID 0x{:016X}.",
           static_cast<uint32_t>(offer_id));
       break;
+    case X_MARKETPLACE_ENTRYPOINT::MembershipList:
+      desc = fmt::format(
+          "Game requested to open marketplace page with all xbox live "
+          "memberships 0x{:016X}.",
+          static_cast<uint32_t>(offer_id));
+      break;
+    case X_MARKETPLACE_ENTRYPOINT::MembershipItem:
+      desc = fmt::format(
+          "Game requested to open marketplace page for an xbox live "
+          "memberships 0x{:016X}.",
+          static_cast<uint32_t>(offer_id));
+      break;
+    case X_MARKETPLACE_ENTRYPOINT::ContentList_Background:
+      // Used when accessing microsoft points
+      desc = fmt::format(
+          "Xbox Marketplace requested access to Microsoft Points offer page "
+          "0x{:016X}.",
+          static_cast<uint32_t>(offer_id));
+      break;
+    case X_MARKETPLACE_ENTRYPOINT::ContentItem_Background:
+      // Used when accessing credit card information and calls
+      // XamShowCreditCardUI
+      desc = fmt::format(
+          "Xbox Marketplace requested access to credit card information page "
+          "0x{:016X}.",
+          static_cast<uint32_t>(offer_id));
```

Comment:
offer_id is 8 bytes, so cast it to uint64_t

Reply:
done